rclcpp/include/rclcpp/strategies/allocator_memory_strategy.hpp

```diff
@@ -504,7 +504,7 @@ class AllocatorMemoryStrategy : public memory_strategy::MemoryStrategy
   using VectorRebind =
     std::vector<T, typename std::allocator_traits<Alloc>::template rebind_alloc<T>>;
 
-  VectorRebind<const rcl_guard_condition_t *> guard_conditions_;
+  VectorRebind<GuardCondition::SharedPtr> guard_conditions_;
```

Comment:
Why does this need to be a shared pointer now? Previously it did not share ownership, is that now necessary?